lib/realtime/block_waiver.ex

```diff
@@ -94,22 +101,25 @@ defmodule Realtime.BlockWaiver do
   defp bank_working([], working), do: [working]
   defp bank_working(acc, working), do: acc ++ [working]
 
-  @spec same_remark?(map(), map()) :: boolean
-  defp same_remark?(%StopTimeUpdate{remark: remark1}, %StopTimeUpdate{remark: remark2}),
-    do: remark1 == remark2
+  @spec same_cause?(StopTimeUpdate.t(), StopTimeUpdate.t()) :: boolean()
```

Comment:
Dialyzer is failing here because it expects this to be `same_cause?(atom() | STU.t(), atom() | STU.t())  :: boolean()`. Any idea why it'd expect that?

Reply:
I don't, but I don't know much about Dialyzer's type inference algorithm and wasn't able to readily find out.

Reply:
It's because concentrates `defstruct_accessors` macro expects you get the fields with accessor functions instead of accessing the fields directly.

Discussion: https://mbta.slack.com/archives/C10EPAU6Q/p1590005263127900